MapboxNavigation/RouteVoiceController.swift

```diff
@@ -181,16 +185,16 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
         return true
     }
     
-    func speak(_ text: String, error: String? = nil) {
+    func speak(_ text: String, error: Error?) {
```

Comment:
As part of #624, I proposed renaming this method:

https://github.com/mapbox/mapbox-navigation-ios/blob/2097f6d5cb95cdfa154d5c56560a8c0ffa8e107e/MapboxNavigation/RouteVoiceController.swift#L196

and having `alertLevelDidChange(_:)` set up the input data for this method instead of storing the input data on instance properties:

https://github.com/mapbox/mapbox-navigation-ios/blob/2097f6d5cb95cdfa154d5c56560a8c0ffa8e107e/MapboxNavigation/RouteVoiceController.swift#L160-L165

PollyVoiceController overrides `speak(for:userDistance:)` instead of `alertLevelDidChange(_:)`:

https://github.com/mapbox/mapbox-navigation-ios/blob/2097f6d5cb95cdfa154d5c56560a8c0ffa8e107e/MapboxNavigation/PollyVoiceController.swift#L51

Overall, PollyVoiceController and RouteVoiceController end up being a lot easier to understand to those who haven’t been involved in writing these classes.